src/main/resources/org/jenkinsci/plugins/scriptsecurity/sandbox/whitelists/generic-whitelist

```diff
@@ -738,11 +739,16 @@ method java.util.Random nextInt
 method java.util.Random nextInt int
 method java.util.Random nextLong
 staticMethod java.util.TimeZone getTimeZone java.lang.String
+new java.util.TreeMap
 new java.util.TreeSet
 staticMethod java.util.UUID randomUUID
+method java.util.concurrent.BlockingQueue poll long java.util.concurrent.TimeUnit
 method java.util.concurrent.Callable call
+new java.util.concurrent.LinkedBlockingQueue
```

Comment:
I'm not sure if it makes sense to add the `BlockingQueue`-related signatures for two main reasons:
* If you use blocking methods like `BlockingQueue.poll` in a Pipeline, the CPS VM thread will be stuck, which will prevent the Pipeline from executing correctly (for example other parallel branches will not be able to progress).
* These signatures seem indicative of using multiple threads from sandboxed scripts, which is a security risk because other threads will execute outside of the sandbox.
 
If you just want a data structure that has queue-like behavior, maybe we could add signatures related to `LinkedList` instead?

Reply:
@gvprathyusha6 should be able to clarify more on this.

Reply:
> These signatures seem indicative of using multiple threads from sandboxed scripts, which is a security risk because other threads will execute outside of the sandbox.

@dwnusbaum I would like to better understand the above concern. What do you mean by "other threads"? We are looking at using one `BlockingQueue` object across multiple `parallel` (as in the Jenkins step) branches. We are not explicitly spawning any threads (we can't anyway, inside the sandbo) and we will not be able to expose this object to an outside thread too. Do you see any concern with this approach?

Reply:
@haridsv I am referring to any code in a Pipeline that spawns Java threads. The `parallel` step does not actually spawn multiple Java threads that run in parallel. See the final paragraph [here](https://github.com/jenkinsci/workflow-cps-plugin#technical-design) for more details, but essentially there is only ever one thread executing the Pipeline at a time, so depending on your exact use case you would probably be fine with a regular list or queue instead of needing a `BlockingQueue` since all of the Groovy code in the parallel branches are effectively synchronized with respect to each other.

Reply:
@dwnusbaum Thank you so much, that makes total sense!

Reply:
@dwnusbaum Per our discussion, I have removed the signatures related to BlockingQueue owing to the suggested alternatives. Please take a look.